### PR TITLE
Licensing Portal: Remove non-primary payment methods modal

### PR DESCRIPTION
**Pull request by vitozev**
#### Context 

- Design: JVpmiQH74FKEIrPLjqv8p7-fi-1262%3A12746

#### Changes proposed in this Pull Request

* This PR makes the Licensing portal uses the new REST API endpoints for removing payment methods in Stripe. With this, we can thoroughly test add/remove payment method flows in the licensing portal.

**Here's a short video overview:**

https://user-images.githubusercontent.com/1749918/156796197-be392553-3ce7-4c23-9339-13a37176f2e1.mp4

#### Testing instructions

**Prerequisites**
- See pbtFFM-1v8-p2

**Instructions**

- If you do not have a partner key, please get in touch with Avalon for one, or you will be unable to view the UI.
- Checkout PR locally, run `yarn && yarn start-jetpack-cloud`.
- Visit http://jetpack.cloud.localhost:3000/partner-portal/payment-methods and select your partner key, if prompted.
- Add multiple payment methods (by reading the prerequisites, you should be able to use Stripe's [test cards available](https://stripe.com/docs/testing))
- Now, test the remove payment method flow from the actions (three dots) menu.
- Verify that:
  - The credit cards are removed from the customer in Stripe (by refreshing the page and see that the item is no longer part of the list)
  - The implementation matches the design.
  - The UI is looking good on different resolutions.
  - No regression is introduced by this PR.

Related to 1201734780767770-as-1201735329399991

### CODE REVIEW

File: client/state/partner-portal/stored-cards/reducer.ts [hunk 1]
```diff
@@ -45,8 +45,45 @@ export const isFetching: Reducer< boolean, IsFetchingAction > = ( state = false,
 	return state;
 };
 
+type IsDeletingAction =
```
Comment (rcanepa) [inline client/state/partner-portal/stored-cards/reducer.ts:48]: I'm not 100% convinced about the name for this type. Could something like `DeletingActionStatus` be clearer? 

Honestly, I don't have a strong opinion on this, but I don't think I ever saw a name representing a boolean state for a type.

**Review by rcanepa — APPROVED** [review]
LGTM.

Code changes look good, and deleting works as described.

File: client/state/partner-portal/stored-cards/reducer.ts (hunk 1 above)
Comment (vitozev) [reply client/state/partner-portal/stored-cards/reducer.ts:48]: Thanks! I agree with this, I'll change this in the second PR for removing primary payment methods.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7170448

Hi @vitozev, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include this string: `Removing card`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7170448).

**Pull request merged**